ocs_ci/deployment/deployment.py

```
@@ -4060,6 +4060,388 @@ def _patch_ramen_csv(self, csv_prefix, namespace, new_image):
+        binary_path = f"{ramenctl_dir}/ramenctl"
+        if not os.path.exists(binary_path):
+            logger.error(
+                f"[ramenctl-e2e] Binary not found at {binary_path} after build"
+            )
+            return False
+
+        # Verify binary by running help command (ramenctl doesn't support --version)
+        verify_cmd = f"cd {ramenctl_dir} && ./ramenctl help"
+        logger.info(f"[ramenctl-e2e] Verifying binary with command: {verify_cmd}")
+        try:
+            help_output = run_cmd(verify_cmd, shell=True)
```

Ruff (0.16.1)

[error] Function call with `shell=True` parameter identified, security issue

(S604)